src/TConsole.cpp

```diff
@@ -1871,11 +1871,20 @@ std::list<int> TConsole::_getFgColor()
         return result;
     }
 
-    if (static_cast<int>(buffer.buffer.at(y).size()) - 1 >= x) {
-        QColor color(buffer.buffer.at(y).at(x).foreground());
+    auto line = buffer.buffer.at(y);
+    int len = static_cast<int>(line.size());
+    if (len - 1 >= x) {
+        int n = 1;
+        QColor color(line.at(x).foreground());
         result.push_back(color.red());
         result.push_back(color.green());
         result.push_back(color.blue());
+        while (len - 1 >= x+n) {
+            if (color != line.at(x+n).foreground())
```

Comment:
Braces around here and below

Reply:
I'm not used to *always* requiring braces …

Reply:
I didn't like it at first - but we looked at Apple's GOTO fail and decided it wasn't worth it. It grows on you eventually.